src/task/executable_task.rs

```diff
@@ -181,6 +185,68 @@ impl<'p> ExecutableTask<'p> {
             stderr: stderr_handle.await.unwrap(),
         })
     }
+
+    pub(crate) async fn can_skip(
+        &self,
+        lock_file: &LockFileDerivedData<'_>,
+    ) -> Result<bool, std::io::Error> {
+        tracing::info!("Checking if task can be skipped");
+        let task_cache_folder = self.project().task_cache_folder();
+
+        let project_name = self.project().name();
+        let environment_name = self.run_environment.name();
+
+        let cache_name = format!(
+            "{}-{}-{}.json",
+            project_name,
+            environment_name,
+            self.name().unwrap_or("default")
+        );
+
+        let cache_file = task_cache_folder.join(cache_name);
+        if cache_file.exists() {
+            let cache = std::fs::read_to_string(&cache_file).unwrap();
+            let cache: TaskCache = serde_json::from_str(&cache).unwrap();
+            let hash = TaskHash::from_task(self, &lock_file.lock_file).await;
+            if let Ok(Some(hash)) = hash {
+                return Ok(hash.computation_hash() == cache.hash);
+            }
+        }
+        Ok(false)
+    }
+
+    pub(crate) async fn save_cache(
+        &self,
+        lock_file: &LockFileDerivedData<'_>,
+    ) -> Result<(), std::io::Error> {
+        let task_cache_folder = self.project().task_cache_folder();
+        if !task_cache_folder.exists() {
+            std::fs::create_dir_all(&task_cache_folder)?;
+        }
+        let project_name = self.project().name();
+        let environment_name = self.run_environment.name();
+
+        let cache_name = format!(
+            "{}-{}-{}.json",
```

Comment:
projects and environments could also contain dashes. Is that a problem in the future?